test/JDBC/expected/TestSpatialFun-vu-verify.out

```diff
@@ -0,0 +1,932 @@
+CASE WHEN PointColumn.STEquals(@referencePoint) <= @isEqual THEN ''Close''
+ELSE ''Far''
+END AS Proximity
+FROM YourTableTemp
+WHERE PointColumn.STEquals(@referencePoint) <= @isEqual;';
+SET @params = N'@referencePoint geometry, @isEqual float';
+EXEC sp_executesql @sql, @params, @referencePoint, @isEqual;
+go
+~~START~~
+int#!#bit#!#text
+1#!#1#!#Close
+2#!#1#!#Close
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT PointColumn.STEquals(@referencePoint) AS Equality FROM YourTableTemp;
+SELECT YourTableTemp.PointColumn.STEquals(@referencePoint) AS Equality FROM YourTableTemp;
+SELECT dbo.YourTableTemp.PointColumn.STEquals(@referencePoint) AS Equality FROM YourTableTemp;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Remote procedure/function reference with 4-part object name is not currently supported in Babelfish)~~
+
+
+DECLARE @pnt geometry;
+SET @pnt = geometry::STGeomFromText('POINT(-122.34900 47.65100)', 4326);
+select geometry::Point(@pnt.STY, @pnt.STX, 4326).STEquals(@pnt)
+go
+~~START~~
+bit
+0
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+DECLARE @equal BIT = 1;
+SELECT ROUND(PointColumn.STEquals(@referencePoint) / @equal, 0) * @equal AS Equalitygroup,
+COUNT(*) AS PointCount
+FROM YourTableTemp
+GROUP BY ROUND(PointColumn.STEquals(@referencePoint) / @equal, 0) * @equal
+ORDER BY Equalitygroup;
+GO
+~~START~~
+numeric#!#int
+1#!#2
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(1.0, 0.0, 4326);
+SELECT ROUND(PointColumn.STEquals(@referencePoint) / @referencePoint.STX, 0) * @referencePoint.STX AS Equalitygroup,
+COUNT(*) AS PointCount
+FROM YourTableTemp
+GROUP BY ROUND(PointColumn.STEquals(@referencePoint) / @referencePoint.STX, 0) * @referencePoint.STX
+ORDER BY Equalitygroup;
+GO
+~~START~~
+float#!#int
+0.0#!#2
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, PointColumn1.STEquals(PointColumn2) AS equal,
+cast(PointColumn1.STEquals(@referencePoint) as float) - LAG(PointColumn1.STEquals(PointColumn2)) OVER (ORDER BY ID) AS Equalitygroup 
+FROM YourTableTemp2 ORDER BY PointColumn1.STX;
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: cannot cast type "bit" to double precision)~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, PointColumn.STEquals(@referencePoint) AS equal,
+cast(@referencePoint.STContains(PointColumn) as float) - LAG(@referencePoint.STX) OVER (ORDER BY ID) AS Equalitygroup
+FROM YourTableTemp ORDER BY PointColumn.STX;
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: cannot cast type "bit" to double precision)~~
+
+
+
+
+
+
+--For STContains
+DECLARE @point1 geometry, @point2 geometry;
+SET @point1 = geometry::STPointFromText('POINT(-122.34900 47.65100)', 4326);
+SET @point2 = geometry::STGeomFromText('POINT(-122.35000 47.65000)', 4326);
+SELECT STContains(@point1, @point2);
```

Comment:
How precise the calculations done by PostGIS and does it match with T-SQL? What about points having big scale, Something like:
1. 
```
SET @point1 = geometry::STPointFromText('POINT(-122.354657658684900 47.658678768678100)', 4326);
SET @point2 = geometry::STPointFromText('POINT(-122.354657658684900 47.658678768678100)', 4326);
SELECT STContains(@point1, @point2);
```
Should output 1.
2. 
```
SET @point1 = geometry::STPointFromText('POINT(-122.354657658684900 47.658678768678100)', 4326);
SET @point2 = geometry::STPointFromText('POINT(-122.354657658684000 47.658678768678100)', 4326);
SELECT STContains(@point1, @point2);
```
Should output 0.

Reply:
Resolved Added Test cases with precision.